### PR TITLE
Java Solution for Jury Marks

### PR DESCRIPTION
**Pull request by brad182**
added java solution with notes included

### CODE REVIEW

File: solutions/cf-831C.mdx [hunk 1]
```diff
@@ -50,5 +50,53 @@ int main() {
...
+	public static void main (String[] args) throws IOException {
+		Kattio io = new Kattio();
+		int k = io.nextInt();
+		int n = io.nextInt();
+		int[] a = new int[k + 1];
+		int[] b = new int[n];
+
+		for (int x = 1; x <= k; x++) { // read k numbers
+			a[x] = io.nextInt();
+			a[x] += a[x - 1];
```
Comment (HarryWangATX) [inline solutions/cf-831C.mdx:70]:
```suggestion
			a[x] = io.nextInt() + a[x - 1];
```

**Review by HarryWangATX — CHANGES_REQUESTED** [review]

**Review by dongliu0426 — APPROVED** [review]

File: solutions/cf-831C.mdx [hunk 2]
```diff
@@ -50,5 +50,52 @@ int main() {
...
+
+public class JuryMarks {
+	public static void main (String[] args) throws IOException {
+		Kattio io = new Kattio();
+		int k = io.nextInt();
+		int n = io.nextInt();
+		int[] a = new int[k + 1];
+		int[] b = new int[n];
+
+		for (int x = 1; x <= k; x++) { // read k numbers
```
Comment (SansPapyrus683) [inline solutions/cf-831C.mdx:68]: uh two spaces after the code thing for inline comments
```suggestion
		for (int x = 1; x <= k; x++) {  // read k numbers
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
hi

File: solutions/cf-831C.mdx [hunk 3]
```diff
@@ -50,5 +50,52 @@ int main() {
...
+		}
+		int answer = 0;
+		for (int x : init) {  // for each number in the set
+			Set<Integer> points = new TreeSet<>();  // create a new set
+			for (int i = 1; i <= numJury; i++) {
+				points.add(x + marks[i]);
+			}
+			boolean flag = true;
+			for (int p : scores) {
+				flag &= points.contains(p);
```
Comment (SansPapyrus683) [inline solutions/cf-831C.mdx:86]: wait bitwise &?
maybe add a comment this seems a little unintuitive

File: solutions/cf-831C.mdx [hunk 4]
```diff
@@ -50,5 +50,52 @@ int main() {
...
+			}
+			if (flag == true) {  // add one if it ends up being true
```
Comment (SansPapyrus683) [inline solutions/cf-831C.mdx:88]: just `if (flag)` should work

File: solutions/cf-831C.mdx [hunk 5]
```diff
@@ -50,5 +50,52 @@ int main() {
...
+import java.io.*;
+import java.util.*;
+
+public class JuryMarks {
+	public static void main(String[] args) throws IOException {
+		Kattio io = new Kattio();
+		int numJury = io.nextInt();
+		int numScores = io.nextInt();
+		int[] marks = new int[numJury + 1];
+		int[] scores = new int[n];
```
Comment (SansPapyrus683) [inline solutions/cf-831C.mdx:66]: did you forget to rename stuff here

File: solutions/cf-831C.mdx [hunk 6]
```diff
@@ -50,5 +50,52 @@ int main() {
...
+		for (int x = 1; x <= numJury; x++) { // add to set
+			init.add(scores[0] - marks[x]);
+		}
+		int answer = 0;
+		for (int x : init) {  // for each number in the set
+			Set<Integer> points = new TreeSet<>();  // create a new set
+			for (int i = 1; i <= numJury; i++) {
+				points.add(x + marks[i]);
+			}
+			boolean flag = true;
```
Comment (SansPapyrus683) [inline solutions/cf-831C.mdx:84]: what's this a flag for

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/cf-831C.mdx (hunk 3 above)
Comment (brad182) [reply solutions/cf-831C.mdx:86]: copied the other implementation for this part for consistency

Comment (SansPapyrus683) [reply solutions/cf-831C.mdx:86]: yeah ok makes sense

File: solutions/cf-831C.mdx [hunk 7]
```diff
@@ -50,5 +50,52 @@ int main() {
...
+			marks[x] = io.nextInt() + marks[x - 1];
+		}
+		for (int x = 0; x < numScores; x++) {  // read n numbers
+			scores[x] = io.nextInt();
+		}
+		Set<Integer> init = new TreeSet<>();
+		for (int x = 1; x <= numJury; x++) { // add to set
+			init.add(scores[0] - marks[x]);
+		}
+		int answer = 0;
```
Comment (SansPapyrus683) [inline solutions/cf-831C.mdx:78]: uh don't think renaming `ans` to `answer` was necessary lmao i'm not that nitpicky

Comment (brad182) [reply solutions/cf-831C.mdx:78]: oh i probably did that unintentionally. but you are one of the strictest reviewers though

**Review by SansPapyrus683 — APPROVED** [review]

**Review by maggie-j-liu — APPROVED** [review]

**Pull request merged**